state/modelgeneration.go

```diff
@@ -0,0 +1,233 @@
+package state
+
+import (
+	"strconv"
+
+	"github.com/juju/juju/core/model"
+
+	"github.com/juju/errors"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+// generationDoc represents the state of a model generation in MongoDB.
+type generationDoc struct {
+	Id string `bson:"generation-id"`
```

Comment:
Do we need a DocID for generationDoc?  What data is the Id here going to contain?